<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>recur_knapsack</class>
 <widget class="QWidget" name="recur_knapsack">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1970</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>1000</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>999999</width>
    <height>1000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>recur_knapsack</string>
  </property>
  <widget class="QPushButton" name="memory_knap">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>50</y>
     <width>201</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>11</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton
{
	font: 11pt &quot;华文楷体&quot;;
}</string>
   </property>
   <property name="text">
    <string>备忘录0-1背包问题</string>
   </property>
  </widget>
  <widget class="QPushButton" name="backtrack_knap">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>50</y>
     <width>201</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>11</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton
{
	font: 11pt &quot;华文楷体&quot;;
}</string>
   </property>
   <property name="text">
    <string>回溯0-1背包问题</string>
   </property>
  </widget>
  <widget class="QPushButton" name="restore">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>50</y>
     <width>191</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>11</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton
{
	font: 11pt &quot;华文楷体&quot;;
}</string>
   </property>
   <property name="text">
    <string>重置</string>
   </property>
  </widget>
  <widget class="QPushButton" name="return_main">
   <property name="geometry">
    <rect>
     <x>980</x>
     <y>50</y>
     <width>171</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>11</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton
{
	font: 11pt &quot;华文楷体&quot;;
}</string>
   </property>
   <property name="text">
    <string>返回</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>690</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>15</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
font: 15pt &quot;华文楷体&quot;;

}</string>
   </property>
   <property name="text">
    <string>STACK</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QTextEdit" name="STACK">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>720</y>
     <width>261</width>
     <height>231</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QTextEdit {
    border-width: 2px 2px 0px 2px; /* 上、右、下边框1px，左边框0px */
    border-style: solid;
    border-color: black; /* 边框颜色为黑色 */
	font: 13pt &quot;宋体&quot;;
}
</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="input_Knap">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>130</y>
     <width>421</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 13pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="inputMask">
    <string/>
   </property>
   <property name="text">
    <string extracomment="二分搜索，请输入初始序列"/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>物体重量以及价值，(w,v)</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="input_Capacity">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>130</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 13pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>总承重</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="input_Num">
   <property name="geometry">
    <rect>
     <x>620</x>
     <y>130</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 13pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>物体数量</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="grids">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>220</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 12pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>grids</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>180</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>15</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
font: 15pt &quot;华文楷体&quot;;

}</string>
   </property>
   <property name="text">
    <string>步数</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>180</y>
     <width>181</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>15</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
font: 15pt &quot;华文楷体&quot;;

}</string>
   </property>
   <property name="text">
    <string>工作纸带空间</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="max_value">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>220</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 12pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>max_value</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="steps">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>220</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
font: 12pt &quot;华文楷体&quot;;
border:2px solid black
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="readOnly">
    <bool>false</bool>
   </property>
   <property name="placeholderText">
    <string>Steps</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>180</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>华文楷体</family>
     <pointsize>15</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
font: 15pt &quot;华文楷体&quot;;

}</string>
   </property>
   <property name="text">
    <string>最大价值</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QTableWidget" name="inputTape">
   <property name="geometry">
    <rect>
     <x>655</x>
     <y>340</y>
     <width>300</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QTableWidegt
{
border:2px solid black;
}</string>
   </property>
   <property name="textElideMode">
    <enum>Qt::ElideMiddle</enum>
   </property>
  </widget>
  <widget class="QLabel" name="picture_turing_2">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>420</y>
     <width>91</width>
     <height>81</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
border:2px solid white
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="picture_arrowhead_3">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>690</y>
     <width>30</width>
     <height>40</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="output_process">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>240</y>
     <width>201</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLineEdit{
border:2px solid black
}</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="picture_arrowhead_2">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>500</y>
     <width>30</width>
     <height>40</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="picture_arrowhead">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>300</y>
     <width>30</width>
     <height>40</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QTableWidget" name="outputTape">
   <property name="geometry">
    <rect>
     <x>655</x>
     <y>730</y>
     <width>301</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
  </widget>
  <widget class="QLabel" name="picture_turing_3">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>610</y>
     <width>91</width>
     <height>81</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
border:2px solid white
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QTableWidget" name="workTape">
   <property name="geometry">
    <rect>
     <x>655</x>
     <y>540</y>
     <width>301</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
  </widget>
  <widget class="QLabel" name="picture_turing">
   <property name="geometry">
    <rect>
     <x>640</x>
     <y>220</y>
     <width>91</width>
     <height>81</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel{
border:2px solid white
}</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
